### PR TITLE
Prevent JVM Segfault

### PR DESCRIPTION
**Pull request by jdye64**
Under certain circumstances importing `dask_cuda` after any `dask_sql` imports will cause an underlying segfault to be triggered in the JVM. In tests thus far adjusting that import order in `context.py` prevents that from happening. Subsequent PRs may be required after further testing.

### CODE REVIEW

**Comment by jdye64** [comment]
I plan to add a pytest with some of the common occurrences where we were seeing the segfault occur so we can guard against those in the future. I'm just compiling that list offline but will post a pytest in the coming days.

**Comment by jdye64** [comment]
rerun tests

**Comment by charlesbluca** [comment]
Cool, looks like this triggers the segfault! Is there any way we can wrap this up in a GPU-only test?

**Comment by jdye64** [comment]
> Cool, looks like this triggers the segfault! Is there any way we can wrap this up in a GPU-only test?

I think it is actually failing to find `dask_cuda` still

**Comment by charlesbluca** [comment]
Checking the gpuCI run, it looks like dask-cuda is there in the conda list step, so I'm assuming it being imported after `dask_sql.Context` is what's causing the segfault here. However it is failing to find the package in the host tests since we don't install any GPU requirements there.

Would we need to add dask-cuda to the host test requirements, or is there a way to contain the conditions for this segfault (i.e. the dask-cuda import) in a test that can be disabled on host?

**Comment by jdye64** [comment]
Ok good news. So I commented out the fix in [commit](https://github.com/dask-contrib/dask-sql/pull/294/commits/ee5162fb4382133b86badc519fb658a4531cb990) and CI failed. This is actually good. Then I un-commented the fix out again in [commit](https://github.com/dask-contrib/dask-sql/pull/294/commits/1ba9a26022126136138ec27f3760291efc8db3af) and it works. Therefore I am satisfied this at least solves the segfault issue we are interested in.

File: dask_sql/context.py [hunk 1]
```diff
@@ -34,6 +34,13 @@
 from dask_sql.physical.rex import RexConverter, core
 from dask_sql.utils import ParsingException
 
+# try:
+#    import dask_cuda  # noqa: F401
+# except ImportError:  # pragma: no cover
+#    pass
+#
```
Comment (ayushdg) [inline dask_sql/context.py:41]: I believe we need to re-add this back prior to jvm startup which happens as a part of line 12.

**Review by ayushdg — CHANGES_REQUESTED** [review]

File: tests/integration/fixtures.py [hunk 2]
```diff
@@ -14,6 +14,16 @@
...
     cudf = None
 
+try:
+    import dask_cudf
+except ImportError:
+    dask_cudf = None
+
+try:
+    from dask_cuda import LocalCUDACluster  # noqa: F401
+except ImportError:
```
Comment (charlesbluca) [inline tests/integration/fixtures.py:24]: Nitpick but can these imports be placed in a specific test? Something like

```python
def test_jvm_segfault():
    import dask_cudf
    from dask_cuda import LocalCUDACluster
```

If they need to be imported here to trigger the segfault, then can they be consolidated with the above try-except block with a comment noting why they appear?

```python
try:
    import cudf
    # importing to check for JVM segfault
    import dask_cudf
    from dask_cuda import LocalCUDACluster
except ImportError:
    cudf = None
```

File: dask_sql/context.py (hunk 1 above)
Comment (jdye64) [reply dask_sql/context.py:41]: Yep, I had forgotten to add it back after testing that commenting it out actually caused errors

File: tests/integration/fixtures.py (hunk 2 above)
Comment (jdye64) [reply tests/integration/fixtures.py:24]: @charlesbluca yes, they needed to be imported there to properly test for the failure. I will combine them however.

File: dask_sql/context.py [hunk 3]
```diff
@@ -34,6 +34,12 @@
 from dask_sql.physical.rex import RexConverter, core
 from dask_sql.utils import ParsingException
 
+try:
+    import dask_cuda  # noqa: F401
+except ImportError:  # pragma: no cover
+    pass
+
+
```
Comment (ayushdg) [inline dask_sql/context.py:42]: These need to be moved prior to line 12 (the first dask_sql import) which ends up starting the jvm.

**Review by ayushdg — CHANGES_REQUESTED** [review]

File: dask_sql/context.py (hunk 3 above)
Comment (jdye64) [reply dask_sql/context.py]: yep, good catch. I got in a hurry and forgot I had changed the ordering for testing.

**Pull request merged**